### PR TITLE
Unguided LDA

### PR DESCRIPTION
**Pull request by psp2**
Unguided LDA Class

### CODE REVIEW

File: clustering/unguided_lda.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
+import sys
+import json
+from collections import Counter
+
+# Make sure to Install NLTK, download stopwords, Gensim, lang_detect, etc.
```
Comment (JoshBurke) [inline clustering/unguided_lda.py]: Put this in repo level README before merge, remove these comments

File: clustering/unguided_lda.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+from langdetect import detect
+
+
+def preprocess_text(text: str):
+    """Tokenize, remove stopwords, and stem words."""
+    processed_text = []
+    stop_words = set(stopwords.words("english"))
+    for token in gensim.utils.simple_preprocess(text):
+        if token not in stop_words:
+            wnl = WordNetLemmatizer()
```
Comment (arynh) [inline clustering/unguided_lda.py]: Does a `WordNetLemmatizer()` need to be constructed every time?

File: clustering/unguided_lda.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+import gensim
+import langdetect
+from nltk.corpus import stopwords
+from nltk.stem import WordNetLemmatizer
+from gensim.utils import simple_preprocess
+from gensim import corpora, models
+from langdetect import detect
+
+
+def preprocess_text(text: str):
```
Comment (arynh) [inline clustering/unguided_lda.py]: `def preprocess_text(text: str): -> list`? (put return type)

File: clustering/unguided_lda.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        """
+        Tests LDA Model by printing out most likely topics for each testing project.
+
+        "Scores" indicate probability of project being part of that topic.
+
+        Keyword Arguments:
+        top_topics -- The number of topics to display, ordered by probability (default 5).
+        """
+
+        # TODO: Figure out how to create and handle testing_data.
```
Comment (arynh) [inline clustering/unguided_lda.py]: todo hould be sfinis

**Review by arynh — CHANGES_REQUESTED** [review]
pls change before merging

**Pull request merged**